src/domain/entities/DataSet.ts

```diff
@@ -0,0 +1,100 @@
+import { Id, Ref } from "$/domain/entities/Ref";
+import { Maybe } from "$/utils/ts-utils";
+import { BasicDataSet, BasicDataSetAttrs, D2Translation } from "$/domain/entities/BasicDataSet";
+
+export interface DataSetAttrs extends BasicDataSetAttrs {
+    name: string;
+    displayFormName: string;
+    formName: string;
+    sections: Section[];
+    dataSetElements: DataSetElement[];
+}
+
+export class DataSet extends BasicDataSet {
+    name: string;
+    displayFormName: string;
+    formName: string;
+    sections: Section[];
+    dataSetElements: DataSetElement[];
+
+    constructor(attrs: DataSetAttrs) {
+        super(attrs);
+        this.name = attrs.name;
+        this.displayFormName = attrs.displayFormName;
+        this.formName = attrs.formName;
+        this.sections = attrs.sections;
+        this.dataSetElements = attrs.dataSetElements;
```

Comment:
super should do all this, except sections.

Reply:
Typescript complains :(